### PR TITLE
Minor changes to alleviate the issue of queue saturation

### PR DESCRIPTION
**Pull request by alvintse1**
Leonard and I realized that, under the current setup of PhxApi, the message queue would become fully saturated after running for a few hours. This is due to the fact that the speed of appending things to the queue is faster than the speed of fetching things from the queue. One of the most likely reasons is that, under the current design of the Roq engine, more than 1000 symbols are subscribed. The speed of consumption from the queue would surely be very slow, as each of the orderbook updates from the 1000+ symbols has to go through "on_order_book_snapshot" or "on_order_book_update", followed by "exec_state_evaluation". This issue is addressed in this PR.

### CODE REVIEW

File: src/phx/api/phx_api.py [hunk 1]
```diff
@@ -79,6 +79,7 @@ def __init__(
         self.config: dict = config or {}
         self.mkt_symbols: Set[Ticker] = set([(exchange, symbol) for symbol in mkt_symbols])
         self.trading_symbols: Set[Ticker] = set([(exchange, symbol) for symbol in trading_symbols])
+        self.set_of_symbol_names = {symbol[1] for symbol in self.mkt_symbols.union(self.trading_symbols)}
```
Comment (eugenek1492) [inline src/phx/api/phx_api.py:82]: `self.mkt_symbols` and `self.trading_symbols` are list of symbols, not list of tuples.
this line should be
self.set_of_symbol_names = {symbol for symbol in self.mkt_symbols.union(self.trading_symbols)}

**Review by eugenek1492 — CHANGES_REQUESTED** [review]
Please run integration  tests on your code changes before next PR update to verify the changes work OK.

File: src/phx/api/phx_api.py [hunk 2]
```diff
@@ -172,7 +173,10 @@ def dispatch(self):
                 # call internal handler
                 match msg:
                     case OrderBookUpdate():
-                        self.on_order_book_update(msg)
+                        if msg.symbol in self.set_of_symbol_names:
+                            self.on_order_book_update(msg)
+                        else:
+                            continue
```
Comment (eugenek1492) [inline src/phx/api/phx_api.py:179]: please add debug message here (even temporary) to verify we get into this statement?

File: src/phx/api/phx_api.py (hunk 1 above)
Comment (alvintse1) [reply src/phx/api/phx_api.py:82]: self.mkt_symbols.union(self.trading_symbols) is a set of tickers: (exchange, symbol). So symbol[1] only fetches the symbol component.

File: src/phx/api/phx_api.py (hunk 2 above)
Comment (alvintse1) [reply src/phx/api/phx_api.py:179]: I just ran an integrated test with debug messages to verify this!

File: src/phx/api/phx_api.py (hunk 1 above)
Comment (eugenek1492) [reply src/phx/api/phx_api.py:82]: Mhy comment was wrong , the current implementation is correct.

**Review by eugenek1492 — APPROVED** [review]
Looks good

**Pull request merged**